src/providers/GlobalProvider/GlobalProvider.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { createContext, useContext, useReducer, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { changeSearch, changeTheme, initialState, reducer, setVideos } from './duck';
+import { theme } from '../../assets/styles/theme';
+
+const GlobalContext = createContext({
+  videos: [],
+  search: '',
+  theme: theme.lightTheme,
+  setVideosCall: () => {},
+  changeSearchCall: () => {},
+});
+
+export const useGlobalContext = () => {
+  const context = useContext(GlobalContext);
+  if (!context) {
+    throw new Error("You can't use useGlobalContext outside a GlobalProvider");
+  }
+  return context;
+};
+
+const GlobalProvider = (props) => {
+  const { children } = props;
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const setVideosCall = useCallback(
+    (videos) => {
+      dispatch(setVideos(videos));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
these disable rules are always™ correct in that we should not use them and rather follow what they recommend.  In these cases we don't need to update the function definition each time any part of `state` changes but rather only the functions it is creating.   While this *works* this might be using resources to re-create the function when it is not needed.